inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/TaskProfileDto.java

```diff
@@ -0,0 +1,548 @@
+        sqlServerJob.setHistory(history);
+
+        return sqlServerJob;
+    }
+
+    public static MqttJob getMqttJob(DataConfig dataConfigs) {
+        MqttJob.MqttJobConfig config = GSON.fromJson(dataConfigs.getExtParams(),
+                MqttJob.MqttJobConfig.class);
+        MqttJob mqttJob = new MqttJob();
+
+        mqttJob.setServerURI(config.getServerURI());
+        mqttJob.setUserName(config.getUsername());
+        mqttJob.setPassword(config.getPassword());
+        mqttJob.setTopic(config.getTopic());
+        mqttJob.setConnectionTimeOut(config.getConnectionTimeOut());
+        mqttJob.setKeepAliveInterval(config.getKeepAliveInterval());
+        mqttJob.setQos(config.getQos());
+        mqttJob.setCleanSession(config.getCleanSession());
+        mqttJob.setClientIdPrefix(config.getClientId());
+        mqttJob.setQueueSize(config.getQueueSize());
+        mqttJob.setAutomaticReconnect(config.getAutomaticReconnect());
+        mqttJob.setMqttVersion(config.getMqttVersion());
+
+        return mqttJob;
+    }
+
+    private static Proxy getProxy(DataConfig dataConfigs) {
+        Proxy proxy = new Proxy();
+        Manager manager = new Manager();
+        AgentConfiguration agentConf = AgentConfiguration.getAgentConf();
+        manager.setHost(agentConf.get(AGENT_MANAGER_VIP_HTTP_HOST));
+        manager.setPort(agentConf.get(AGENT_MANAGER_VIP_HTTP_PORT));
+        proxy.setInlongGroupId(dataConfigs.getInlongGroupId());
+        proxy.setInlongStreamId(dataConfigs.getInlongStreamId());
+        proxy.setManager(manager);
+        if (null != dataConfigs.getSyncSend()) {
+            proxy.setSync(dataConfigs.getSyncSend() == SYNC_SEND_OPEN);
+        }
+        if (null != dataConfigs.getSyncPartitionKey()) {
+            proxy.setPartitionKey(dataConfigs.getSyncPartitionKey());
+        }
+        return proxy;
+    }
+
+    /**
+     * convert DataConfig to TaskProfile
+     */
+    public static TaskProfile convertToTaskProfile(DataConfig dataConfig) {
+        if (!dataConfig.isValid()) {
+            throw new IllegalArgumentException("input dataConfig" + dataConfig + "is invalid please check");
+        }
+
+        TaskProfileDto profileDto = new TaskProfileDto();
+        Proxy proxy = getProxy(dataConfig);
+        profileDto.setProxy(proxy);
+        Task task = new Task();
+
+        // common attribute
+        task.setId(String.valueOf(dataConfig.getTaskId()));
+        task.setGroupId(dataConfig.getInlongGroupId());
+        task.setStreamId(dataConfig.getInlongStreamId());
+        task.setChannel(DEFAULT_CHANNEL);
+        task.setIp(dataConfig.getIp());
+        task.setOp(dataConfig.getOp());
+        task.setDeliveryTime(dataConfig.getDeliveryTime());
+        task.setUuid(dataConfig.getUuid());
+        task.setVersion(dataConfig.getVersion());
+        task.setState(dataConfig.getState());
+
+        // set sink type
+        if (dataConfig.getDataReportType() == 0) {
```

Comment:
0 and 1 can be constant with meaningful name , the best way is making type a enum

Reply:
ok